src/Symfony/Component/DependencyInjection/Argument/ServiceLocatorArgument.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Represents a service locator able to lazy load a given range of services.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ *
+ * @experimental in version 3.3
+ */
+class ServiceLocatorArgument implements ArgumentInterface
+{
+    private $values;
+
+    /**
+     * @param array $values An array of mixed entries indexed by identifier
```

Comment:
Please see https://github.com/symfony/symfony/pull/21553#discussion_r99926055. There's no explicit notion of parameters here,  only services to `get()` but services can be any value. I have no strong opinion about, but that's PSR-11 compliant.